### PR TITLE
Update BuildFullResourceName to validate for non empty arguments. Closes #3601

### LINKED ISSUE #3601: `steampipeconfig/modconfig/BuildFullResourceName` should validate for non empty arguments

### PR DESCRIPTION
**Pull request by pskrbasu**

### CODE REVIEW

File: pkg/dashboard/dashboardexecute/dashboard_execution_tree.go [hunk 1]
```diff
@@ -3,11 +3,12 @@ package dashboardexecute
 import (
 	"context"
 	"fmt"
-	"golang.org/x/exp/maps"
 	"log"
 	"sync"
 	"time"
 
+	"golang.org/x/exp/maps"
+
```
Comment (binaek) [inline pkg/dashboard/dashboardexecute/dashboard_execution_tree.go:11]: line not required

File: pkg/steampipeconfig/modconfig/dashboard.go [hunk 2]
```diff
@@ -2,11 +2,12 @@ package modconfig
 
 import (
 	"fmt"
+	"strings"
+
 	"github.com/spf13/viper"
 	typehelpers "github.com/turbot/go-kit/types"
 	"github.com/turbot/steampipe/pkg/constants"
 	"github.com/zclconf/go-cty/cty"
-	"strings"
```
Comment (binaek) [inline pkg/steampipeconfig/modconfig/dashboard.go:11]: remove blank line

File: pkg/steampipeconfig/modconfig/parsed_name.go [hunk 3]
```diff
@@ -44,21 +44,43 @@ func (p *ParsedResourceName) ToResourceName() string {
 	return BuildModResourceName(p.ItemType, p.Name)
 }
 
-func (p *ParsedResourceName) ToFullName() string {
-	return BuildFullResourceName(p.Mod, p.ItemType, p.Name)
+func (p *ParsedResourceName) ToFullName() (string, error) {
+	fullName, err := BuildFullResourceName(p.Mod, p.ItemType, p.Name)
+	if err != nil {
```
Comment (binaek) [inline pkg/steampipeconfig/modconfig/parsed_name.go:49]: error check not required - just do
`return BuildFullResourceName...`

File: pkg/steampipeconfig/modconfig/parsed_name.go [hunk 4]
```diff
@@ -44,21 +44,43 @@ func (p *ParsedResourceName) ToResourceName() string {
...
+	}
+	return fullName, nil
 }
-func (p *ParsedResourceName) ToFullNameWithMod(mod string) string {
+
+func (p *ParsedResourceName) ToFullNameWithMod(mod string) (string, error) {
 	if p.Mod != "" {
-		return p.ToFullName()
+		fullName, err := p.ToFullName()
+		if err != nil {
```
Comment (binaek) [inline pkg/steampipeconfig/modconfig/parsed_name.go:58]: error check not required - see above

File: pkg/steampipeconfig/modconfig/parsed_name.go [hunk 5]
```diff
@@ -44,21 +44,43 @@ func (p *ParsedResourceName) ToResourceName() string {
...
+			return "", err
+		}
+		return fullName, nil
+	}
+	fullResourceName, err := BuildFullResourceName(mod, p.ItemType, p.Name)
+	if err != nil {
```
Comment (binaek) [inline pkg/steampipeconfig/modconfig/parsed_name.go:64]: error check not required

**Review by binaek — CHANGES_REQUESTED** [review]

File: pkg/dashboard/dashboardexecute/snapshot.go [hunk 6]
```diff
@@ -38,7 +38,7 @@ func GenerateSnapshot(ctx context.Context, target string, initData *initialisati
 		return nil, err
 	case snapshot = <-resultChannel:
 		// set the filename root of the snapshot
-		snapshot.FileNameRoot = parsedName.ToFullNameWithMod(w.Mod.ShortName)
+		snapshot.FileNameRoot, err = parsedName.ToFullNameWithMod(w.Mod.ShortName)
```
Comment (kaidaguerre) [inline pkg/dashboard/dashboardexecute/snapshot.go:41]:
```
// set the filename root of the snapshot
		fileRootName, err := parsedName.ToFullNameWithMod(w.Mod.ShortName)
		if err != nil{
			return nil, err
		}
		snapshot.FileNameRoot = fileRootName
		
```

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

**Review by kaidaguerre — APPROVED** [review]

**Pull request merged**